monai/data/dataset.py

```diff
@@ -93,22 +95,38 @@ def __init__(self, data, transform, cache_num=sys.maxsize, cache_rate=1.0):
                 will take the minimum of (cache_num, data_length x cache_rate, data_length).
             cache_rate (float): percentage of cached data in total, default is 1.0 (cache all).
                 will take the minimum of (cache_num, data_length x cache_rate, data_length).
+            num_workers (int): the number of worker processes to use.
+                If 0 a single thread will be used. Default is 0.
         """
         if not isinstance(transform, Compose):
             transform = Compose(transform)
         super().__init__(data, transform)
         self.cache_num = min(cache_num, int(len(self) * cache_rate), len(self))
         self._cache = list()
         print('Load and cache transformed data...')
-        for i in range(self.cache_num):
-            process_bar(i + 1, self.cache_num)
-            item = data[i]
-            for _transform in transform.transforms:
-                # execute all the deterministic transforms before the first random transform
-                if isinstance(_transform, Randomizable):
-                    break
-                item = apply_transform(_transform, item)
-            self._cache.append(item)
+        if num_workers > 0:
+            def _update_bar(i):
+                process_bar(i + 1, self.cache_num)
```

Comment:
If calling `process_bar` in multi-processing, I think the `i` is not the exact count of cached items. So maybe you need to figure out a method to count the cached number and continuously update the bar in the main process.
Thanks.

Reply:
Oh true, thanks for pointing it out.